### PR TITLE
Home work 4

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/PetOwner.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+public class PetOwner {
+	private String name;
+	private String surName;
+	private String petName;
+	private int petAge;
+	private int petCount;
+	private String typeOfPet;
+	private String genderOfPet;
+
```
Comment (Sofasmile) [inline src/PetOwner.java]: add private constructor

File: src/SquareRootFinder.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+public class SquareRootFinder {
+	public double sqrt(double x) {
+		if (x < 0) {
+			throw new IllegalArgumentException("Expected number, bigger than 0");
+		}else {
```
Comment (Sofasmile) [inline src/SquareRootFinder.java]: remove  else. It is unnecessary 
if x<0 you will have Exception

File: untitled/src/AsciiCharSequence.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Override
+	public char charAt(int index) {
+		return (char) stringReplace[index];
+	}
+
+	@Override
+	public CharSequence subSequence(int start, int end) {
+		byte[] result = new byte[end - start];
+		if (end - start >= 0) System.arraycopy(stringReplace, start, result, start - start, end - start);
```
Comment (Sofasmile) [inline untitled/src/AsciiCharSequence.java]: CS

**Review by Sofasmile — COMMENTED** [review]
review only tasks from hw 4

File: untitled/src/AsciiCharSequence.java (hunk 3 above)
Comment (UnitOmega13) [reply untitled/src/AsciiCharSequence.java]: Fixed

**Review by Sofasmile — COMMENTED** [review]
where is this task https://stepik.org/lesson/14513/step/9?unit=4147?

**Comment by UnitOmega13** [comment]
All addition task from Homework 4 created in Homework 3: https://github.com/UnitOmega13/HomeWork/pull/2

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**